### PR TITLE
Send metering pingback only on metering requests

### PR DESCRIPTION
**Pull request by elijahsoria**
Adds a check in the entitlements pingback methods that checks if the request URL has valid GAA params. Does this by modifying and using the urlContainsFreshGaaParams which checks if all necessary elements are present. Adds tests.

### CODE REVIEW

File: src/runtime/entitlements-manager.js
Comment (ChrisAntaki) [inline src/runtime/entitlements-manager.js:204]:
```suggestion
    const nonce = parseQueryString(this.win_.location.search)['gaa_n'];
```

Comment (ChrisAntaki) [reply src/runtime/entitlements-manager.js:204]: This would be a bit more descriptive

File: src/runtime/entitlements-manager.js
Comment (ChrisAntaki) [inline src/runtime/entitlements-manager.js:263]: Could you rename this to `nonce` or `optionalNonce`? https://google.github.io/styleguide/jsguide.html#naming-parameter-names

File: src/runtime/entitlements-manager.js
Comment (ChrisAntaki) [reply src/runtime/entitlements-manager.js:204]: I also suggest this change on line 248

File: src/utils/gaa.js
Comment (ChrisAntaki) [inline src/utils/gaa.js:305]: Could you rename this to `optionalQueryString`? https://google.github.io/styleguide/jsguide.html#naming-parameter-names

File: src/utils/gaa.js
Comment (ChrisAntaki) [inline src/utils/gaa.js:309]:
```suggestion
  const queryString = optionalQueryString || GaaMeteringRegwall.location_.search;
```

Comment (ChrisAntaki) [reply src/utils/gaa.js:309]: Actually, I'm going to send you a PR that refactors this method a bit... please feel free to hold off on changing it

File: src/runtime/entitlements-manager.js
Comment (ChrisAntaki) [inline src/runtime/entitlements-manager.js:216]: This name is much more descriptive, thank you!

File: src/utils/gaa.js
Comment (ChrisAntaki) [reply src/utils/gaa.js:309]: Here's the PR! https://github.com/subscriptions-project/swg-js/pull/1480

File: src/runtime/entitlements-manager.js
Comment (elijahsoria) [reply src/runtime/entitlements-manager.js:263]: optionalNonce it is.

File: src/utils/gaa.js
Comment (elijahsoria) [reply src/utils/gaa.js:305]: Resolving since function is updated.

File: src/utils/gaa.js
Comment (elijahsoria) [reply src/utils/gaa.js:309]: Thanks! Resolving since function is updated.

File: src/runtime/entitlements-manager-test.js
Comment (ChrisAntaki) [inline src/runtime/entitlements-manager-test.js:898]: I don't think this line is needed

File: src/runtime/entitlements-manager-test.js
Comment (ChrisAntaki) [inline src/runtime/entitlements-manager-test.js:1752]: Just curious, why is this needed?

Comment (ChrisAntaki) [inline src/runtime/entitlements-manager-test.js:1752]: `setTimeout` should return a `Number`. Just curious though, is this line still needed?
```suggestion
      sandbox.stub(win, 'setTimeout').returns(1);
```

File: src/runtime/entitlements-manager-test.js
Comment (elijahsoria) [reply src/runtime/entitlements-manager-test.js:898]: You're right! Removing all calls to reset.

File: src/runtime/entitlements-manager-test.js
Comment (elijahsoria) [reply src/runtime/entitlements-manager-test.js:1752]: Changed to returns(1). Without a stub, setTimeout will fail with a `TypeError: Illegal invocation`, I think because it hits L449 on entitlements-manager.js and doesn't know what to do without an implementation of setTimeout?

**Review by ChrisAntaki — APPROVED** [review]

**Comment by mborof** [comment]
Can't our boq actions see the requesting browser tab's query string and pull out the nonce directly?

File: src/runtime/entitlements-manager.js
Comment (jpettitt) [reply src/runtime/entitlements-manager.js]: per Dan `nonce` is officially deprecated it's now `gaaToken` or in this case `optionalGaaToken`

**Review by jpettitt — APPROVED** [review]
LGTM modulo nonce name change

**Pull request merged**
